web/gds-user-ui/src/modules/dashboard/Certificate/registration.tsx

```diff
@@ -0,0 +1,153 @@
+import Card from 'components/ui/Card';
+import TestNetCertificateProgressBar from 'components/TestnetProgress/TestNetCertificateProgressBar.component';
+import FormButton from 'components/ui/FormButton';
+import useCertificateStepper from 'hooks/useCertificateStepper';
+import { FormProvider, useForm, useFormState } from 'react-hook-form';
+
+import { yupResolver } from '@hookform/resolvers/yup';
+import { DevTool } from '@hookform/devtools';
+import { RootStateOrAny, useSelector } from 'react-redux';
+import _ from 'lodash';
+import { hasStepError } from '../../../utils/utils';
+
+import {
+  fieldNamesPerSteps,
+  certificateRegistrationValidationSchema,
+  getCertificateRegistrationDefaultValue
+} from './lib';
+
+const fieldNamesPerStepsEntries = () => Object.entries(fieldNamesPerSteps);
+
+const Certificate: React.FC = () => {
+  const { nextStep, previousStep } = useCertificateStepper();
+  const currentStep: number = useSelector((state: RootStateOrAny) => state.stepper.currentStep);
+  const lastStep: number = useSelector((state: RootStateOrAny) => state.stepper.lastStep);
+  const steps: number = useSelector((state: RootStateOrAny) => state.stepper.steps);
+  const hasReachSubmitStep: boolean = useSelector(
+    (state: RootStateOrAny) => state.stepper.hasReachSubmitStep
+  );
+  const toast = useToast();
+  const current = currentStep === lastStep ? lastStep - 1 : currentStep;
+
+  const methods = useForm({
+    defaultValues: getCertificateRegistrationDefaultValue(),
+    resolver: yupResolver(certificateRegistrationValidationSchema),
+    mode: 'onChange'
+  });
+
+  function getFieldValue(name: string) {
+    return _.get(methods.getValues(), name);
+  }
+
+  function isFormCompleted() {
+    const fieldsNames = fieldNamesPerStepsEntries()[current - 1][1];
+    return fieldsNames.every((n) => getFieldValue(n).toString());
+  }
+
+  function getCurrentFormValue() {
+    const fieldsNames = fieldNamesPerStepsEntries()[current - 1][1];
+    return fieldsNames.reduce((acc, n) => ({ ...acc, [n]: getFieldValue(n) }), {});
+  }
+
+  function hasErroredField() {
+    const fieldsNames = fieldNamesPerStepsEntries()[current - 1][1];
+    return fieldsNames.some((n: any) => methods.getFieldState(n).error);
+  }
+
+  function handleNextStepClick() {
+    if (currentStep === lastStep) {
+      if (hasStepError(steps)) {
+        toast({
+          position: 'top',
+          title: `Please fill all the required fields before submitting`,
+          status: 'error',
+          isClosable: true,
+          containerStyle: {
+            width: '800px',
+            maxWidth: '100%'
+          }
+        });
+      }
+    }
+    if (hasErroredField()) {
+      // i think we should not use alert here , but we need to find a way to display the error message
+      // eslint-disable-next-line no-alert
+      if (window.confirm('Some requirement elements are missing, Would you like to continue?')) {
```

Comment:
@rebeccabilbro same here

Reply:
How about:

```Some elements required for registration are missing; continue anyway?```

I'm guessing this will be used by people who want to progress to the next step of the form, but with the understanding that they will need to come back (perhaps with a technical or legal support aid) to ensure all the required fields are filled before submission.